### PR TITLE
Aaron/add more checksfor mesh factory

### PR DESCRIPTION
**Pull request by Blyron**
#### Description

Add checks for new types of roads to enable generate "road" for parking lanes and "sidewalk" for "road" lanes

Fixes #  

#### Where has this been tested?

  * **Platform(s):** ...
  * **Python version(s):** ...
  * **Unreal Engine version(s):** ...

#### Possible Drawbacks

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/carla/7729)

### CODE REVIEW

File: LibCarla/source/carla/road/MeshFactory.cpp [hunk 1]
```diff
@@ -1131,6 +1170,32 @@ std::map<road::Lane::LaneType , std::vector<std::unique_ptr<Mesh>>> MeshFactory:
...
+  uint32_t MeshFactory::SelectVerticesInWidth(uint32_t default_num_vertices, road::Lane::LaneType type)
+  {
+    switch(type)
+    {
+      case road::Lane::LaneType::Driving:
+      case road::Lane::LaneType::Parking:
+      case road::Lane::LaneType::Bidirectional:
+      {
+        return default_num_vertices;
+        break;
```
Comment (xavisolesoft) [inline LibCarla/source/carla/road/MeshFactory.cpp:1182]:
```suggestion
```

Return code after return is never executed (unreachable) and return already breaks the switch.

File: LibCarla/source/carla/road/MeshFactory.cpp [hunk 2]
```diff
@@ -1131,6 +1170,32 @@ std::map<road::Lane::LaneType , std::vector<std::unique_ptr<Mesh>>> MeshFactory:
...
+      }
+      case road::Lane::LaneType::Shoulder:
+      case road::Lane::LaneType::Sidewalk:
+      case road::Lane::LaneType::Biking:
+      {
+        return 6;
+        break;
```
Comment (xavisolesoft) [inline LibCarla/source/carla/road/MeshFactory.cpp:1189]:
```suggestion
```
Return code after return is never executed (unreachable) and return already breaks the switch.

File: LibCarla/source/carla/road/MeshFactory.cpp [hunk 3]
```diff
@@ -1131,6 +1170,32 @@ std::map<road::Lane::LaneType , std::vector<std::unique_ptr<Mesh>>> MeshFactory:
...
+      }
+      default:
+      {
+        return 2;
+        break;
```
Comment (xavisolesoft) [inline LibCarla/source/carla/road/MeshFactory.cpp:1194]:
```suggestion
```

Return code after return is never executed (unreachable) and return already breaks the switch.

File: LibCarla/source/carla/road/MeshFactory.cpp [hunk 4]
```diff
@@ -725,9 +753,20 @@ std::map<road::Lane::LaneType , std::vector<std::unique_ptr<Mesh>>> MeshFactory:
...
+            case road::Lane::LaneType::Bidirectional:
+            {
+              GenerateLaneMarksForNotCenterLine(lane_section, lane.second, inout, outinfo);
+              outinfo.push_back("white");
+              break;
+            }
+            default:
+            {
+              break;
+            }
```
Comment (xavisolesoft) [inline LibCarla/source/carla/road/MeshFactory.cpp:769]:
```suggestion
```

This code does nothing.

**Comment by Blyron** [comment]
Applied!

**Review by xavisolesoft — APPROVED** [review]

**Pull request merged**